### PR TITLE
*: add a page and related description for the Optimizer Fix Control

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

As the title says, add a page add related description for the Optimizer Fix Control.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/issues/43169 
https://github.com/pingcap/docs-cn/pull/13733

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-fix-controls.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: Optimizer Fix Controls
+summary: 了解 Optimizer Fix Controls 以及如何使用 tidb_opt_fix_control 细粒度地控制 TiDB 优化器的行为
+aliases: ['/docs-cn/dev/optimizer-fix-controls/']
```
Comment (hfxsd) [inline optimizer-fix-controls.md:4]: 新文档不需要 aliases 
```suggestion
summary: 了解 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为
```

File: optimizer-fix-controls.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致执行计划回退等非预期情况的风险：
```
Comment (hfxsd) [inline optimizer-fix-controls.md:9]:
```suggestion
TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致以下非预期情况的风险：
```

File: optimizer-fix-controls.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
+因此我们提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:14]:
```suggestion
因此 TiDB 提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```

File: optimizer-fix-controls.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 是否允许在缺少 [GlobalStats](/statistics.md#动态裁剪模式下的分区表统计信息) 的情况下使用[动态裁剪模式](/partitioned-table.md#动态裁剪模式)访问分区表。
+
+### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">从 v7.1.1 版本开始引入</span>
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 对形如 `c = 10 and (a = 'xx' or (a = 'kk' and b = 1)` 的过滤条件，是否尝试为 `IndexRangeScan` 更加完整地构造扫描范围，即 `range`。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:36]: 是不是应该加上个有括号？像下面这样。PTAL @time-and-fate 
```suggestion
- 对形如 `c = 10 and (a = 'xx' or (a = 'kk' and b = 1))` 的过滤条件，是否尝试为 `IndexRangeScan` 更加完整地构造扫描范围，即 `range`。
```

File: optimizer-fix-controls.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+title: Optimizer Fix Controls
+summary: 了解 Optimizer Fix Controls 以及如何使用 tidb_opt_fix_control 细粒度地控制 TiDB 优化器的行为
+aliases: ['/docs-cn/dev/optimizer-fix-controls/']
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致执行计划回退等非预期情况的风险：
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:11]: 这条看起来不像是风险？如果是风险，需要调整措辞

File: optimizer-fix-controls.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
+因此我们提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
+
+一个 Fix 通常以一个数字编号表示，这个数字编号通常对应一个 GitHub Issue，在 Issue 中会有对技术细节的描述。例如 Fix `44262` 对应 [Issue 44262](https://github.com/pingcap/tidb/issues/44262)。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:16]:
```suggestion
一个 Fix 以一个数字编号表示，该数字编号对应一个 GitHub Issue，在 Issue 中会有对技术细节的描述。例如 Fix `44262` 对应 [Issue 44262](https://github.com/pingcap/tidb/issues/44262)。
```

File: optimizer-fix-controls.md (hunk 5 above)
Comment (time-and-fate) [reply optimizer-fix-controls.md]: 增加了一点描述。PTAL~

**Review by hfxsd — APPROVED** [review]

File: optimizer-fix-controls.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+---
+title: Optimizer Fix Controls
+summary: 了解 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致以下非预期情况的风险：
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。有的行为改变，能在大多数场景下带来改进，但同时可能在极少数场景下导致回退。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:10]:
```suggestion
- 对于某些实现细节，不同的使用场景更适合不同的行为。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
```

File: optimizer-fix-controls.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+summary: 了解 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致以下非预期情况的风险：
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
```
Comment (hfxsd) [inline optimizer-fix-controls.md:12]:
```suggestion

## `tidb_opt_fix_control` 介绍

```

File: optimizer-fix-controls.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+因此 TiDB 提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:13]:
```suggestion
从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```

File: optimizer-fix-controls.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+summary: 了解 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致以下非预期情况的风险：
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
```
Comment (ran-huang) [inline optimizer-fix-controls.md:12]: 建议加一句话总体介绍本文档的内容，比如：

```suggestion

因此，TiDB 提供了 Optimizer Fix Controls 功能，允许用户通过设置一系列 Fix 控制 TiDB 优化器的行为细节。本文档描述了 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为。

```

File: optimizer-fix-controls.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+## `tidb_opt_fix_control` 介绍
+
+从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
+
+一个 Fix 以一个数字编号表示，该数字编号对应一个 GitHub Issue，在 Issue 中会有对技术细节的描述。例如 Fix `44262` 对应 [Issue 44262](https://github.com/pingcap/tidb/issues/44262)。
```
Comment (ran-huang) [inline optimizer-fix-controls.md:17]: 能否一句话介绍什么是 Fix？比如 "Fix 是 TiDB 优化器中某些特定行为的编号，一个 Fix 对应着一个具体的技术细节问题。"

File: optimizer-fix-controls.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
+---
+title: Optimizer Fix Controls
```
Comment (ran-huang) [inline optimizer-fix-controls.md:2]: 这个功能可以翻译成中文吗？

Comment (time-and-fate) [reply optimizer-fix-controls.md:2]: 这个名字是借用了其它数据库里面同样的名字，没想出比较好的翻译。感觉可能类似 Optimizer Hints，直接用英文比较好？
@songrijie 可以帮忙看下吗？

File: optimizer-fix-controls.md (hunk 11 above)
Comment (time-and-fate) [reply optimizer-fix-controls.md]: 已更新，PTAL~

File: optimizer-fix-controls.md (hunk 12 above)
Comment (songrijie) [reply optimizer-fix-controls.md:2]: 就留着英文比较好， 其他数据库也是相同的叫法。

File: optimizer-fix-controls.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+summary: 了解 Optimizer Fix Controls 以及如何使用 `tidb_opt_fix_control` 细粒度地控制 TiDB 优化器的行为
+---
+
+# Optimizer Fix Controls
+
+TiDB 优化器的实现包含很多行为细节。随着产品演进，这些细节往往会发生变化。通常，这些变化都是优化器的改进，但有时仍会有导致以下非预期情况的风险：
+
+- 对于某些实现细节，不同的使用场景更适合不同的行为。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
```
Comment (songrijie) [inline optimizer-fix-controls.md:10]:
```suggestion
- 部分行为的效果和场景相关。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
```

**Review by songrijie — APPROVED** [review]
rest LGTM

File: optimizer-fix-controls.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+因此，TiDB 提供了 Optimizer Fix Controls 功能，允许用户通过设置一系列 Fix 控制 TiDB 优化器的行为细节。本文档介绍了 Optimizer Fix Controls 及其使用方法，并列举了当前 TiDB 支持调整的所有 Fix。
+
+## `tidb_opt_fix_control` 介绍
+
+从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
+
+一个 Fix 是用于调整 TiDB 优化器中一处行为的控制项。它以一个数字编号表示，该数字编号对应一个 GitHub Issue，在 Issue 中会有对技术细节的描述。例如 Fix `44262` 对应 [Issue 44262](https://github.com/pingcap/tidb/issues/44262)。
+
+该变量支持设置多个 Fix，不同 Fix 之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`，其中 `<#issueN>` 代表 Fix 编号。例如：
```
Comment (ran-huang) [inline optimizer-fix-controls.md:21]:
```suggestion
`tidb_opt_fix_control` 支持设置多个 Fix，不同 Fix 之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`，其中 `<#issueN>` 代表 Fix 编号。例如：
```

**Review by ran-huang — APPROVED** [review]
LGTM

File: optimizer-fix-controls.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+`tidb_opt_fix_control` 支持设置多个 Fix，不同 Fix 之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`，其中 `<#issueN>` 代表 Fix 编号。例如：
+
+```sql
+SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
+```
+
+## Optimizer Fix Controls 参考
+
+### [`44262`](https://github.com/pingcap/tidb/issues/44262) <span class="version-mark">从 v7.1.1 版本开始引入</span>
```
Comment (ran-huang) [inline optimizer-fix-controls.md:29]: 这里是不是应该改成 v7.2.0 引入，在 7.1 的 cherry-pick 里再写 v7.1.1 引入？

Comment (qiancai) [reply optimizer-fix-controls.md:29]: +1

Comment (xuyifangreeneyes) [reply optimizer-fix-controls.md:29]: 是的

Comment (time-and-fate) [reply optimizer-fix-controls.md]: 已更新。

**Pull request merged**